bzlib.cabal

```diff
@@ -24,6 +24,12 @@ extra-source-files: cbits/bzlib_private.h cbits/LICENSE
 
 tested-with: GHC==9.8.2, GHC==9.6.4, GHC==9.4.8, GHC==9.2.8, GHC==9.0.2, GHC==8.10.7, GHC==8.8.4, GHC==8.6.5, GHC==8.4.4, GHC==8.2.2, GHC==8.0.2, GHC==7.10.3, GHC==7.8.4, GHC==7.6.3, GHC==7.4.2, GHC==7.2.2, GHC==7.0.4
 
+flag system-bzlib
+    description:
+        Use system bzlib rather than the bundled sources
+    default: True
+    manual: True
```

Comment:
This is the right default, imo.

Using bundled C sources should be the last resort. Even under windows you can build against bzip2 just fine: https://packages.msys2.org/base/mingw-w64-bzip2

Unfortunately, upstream does not provide a pkg-config file, so we can't use the auto fallback mechanism.

Reply:
I understand your position, but my experience with other packages is that practically it would force Windows users to stay on the older, vulnerable version of `bzlib`.

Reply:
Well, feel free to cherry-pick the first commit then, which keeps the old behavior.

Reply:
Thanks, I'll merge and revert the second commit then.